bin/main.ml

```diff
@@ -1,14 +1,22 @@
 let with_open_in fn f =
   let ic = open_in fn in
   match f ic with
-  | r -> close_in ic; r
-  | exception e -> close_in_noerr ic; raise e
+  | r           ->
+      close_in ic;
+      r
+  | exception e ->
```

Comment:
Everything looks good to me, except for the aligned arrows. I don't feel strongly about it, so if you really prefer it that way, I'm ok with it.

Having said that, the rule is either not applied consistently or I don't understand when it should apply.

Reply:
I am ok either way. I agree it's awkward in these cases, but I tend to feel like it pays off when you have lots of similarly sized matches.

I'm not sure what's up with the inconsistency!

I've removed that rule. Let me know if you like this better :)

Reply:
I prefer it this way, but I'm ok with the other way if we figure out why it's inconsistent.